services/QuillLMS/app/models/sales_form_submission.rb

```diff
@@ -30,18 +32,192 @@ class SalesFormSubmission < ApplicationRecord
     QUOTE_REQUEST_TYPE = 'quote request',
     RENEWAL_REQUEST_TYPE = 'renewal request'
   ]
+  FORM_SOURCE = "Form"
+  INTERCOM_SOURCE = "Intercom"
+
+  VITALLY_DISTRICTS_TYPE = "organizations"
+  VITALLY_SCHOOLS_TYPE = "accounts"
+  VITALLY_USERS_TYPE = "users"
+  VITALLY_SALES_FORMS_TYPE = "projects"
+
+  FALLBACK_SCHOOL_NAME = "Unknown School"
+  FALLBACK_DISTRICT_NAME = "Unknown District"
+
+  SCHOOL_QUOTE_REQUEST_TEMPLATE_ID = ENV['VITALLY_SCHOOL_QUOTE_TEMPLATE_ID']
+  DISTRICT_QUOTE_REQUEST_TEMPLATE_ID = ENV['VITALLY_DISTRICT_QUOTE_TEMPLATE_ID']
+  SCHOOL_RENEWAL_REQUEST_TEMPLATE_ID = ENV['VITALLY_SCHOOL_RENEWAL_TEMPLATE_ID']
+  DISTRICT_RENEWAL_REQUEST_TEMPLATE_ID = ENV['VITALLY_DISTRICT_RENEWAL_TEMPLATE_ID']
 
   validates :first_name, presence: true
   validates :last_name, presence: true
   validates :email, presence: true
   validates_email_format_of :email, message: :invalid
-  validates :phone_number, presence: true
-  validates :zipcode, presence: true
-  validates :school_name, presence: true, unless: :district_name
-  validates :district_name, presence: true, unless: :school_name
   validates :school_premium_count_estimate, presence: true, numericality: { greater_than_or_equal_to: 0 }
   validates :teacher_premium_count_estimate, presence: true, numericality: { greater_than_or_equal_to: 0 }
   validates :student_premium_count_estimate, presence: true, numericality: { greater_than_or_equal_to: 0 }
   validates :collection_type, presence: true, inclusion: { in: COLLECTION_TYPES }
   validates :submission_type, presence: true, inclusion: { in: SUBMISSION_TYPES }
+
+  after_save :vitally_callbacks
```

Comment:
I tend to avoid after_save callbacks except those managing state internal to the object.  (e.g. see this [post](https://samuelmullen.com/2013/05/the-problem-with-rails-callbacks/)).  Maybe discuss offline pros/cons?

Reply:
For posterity: Brendan and I had a call about this and basically decided to compromise.  We want the `after_save` because we want to guarantee that this data syncs to Vitally no matter how we end up creating these records.  However, we also definitely don't want to tightly couple the ability to save models to external APIs (which is what happens if you do this synchronously).  So we're keeping `after_save`, but moving the sync work into a Sidekiq worker that's triggered by said `after_save` hook.